### DISCUSSION TITLE
Powershell Script Block Parsing - Winlogbeat

### DISCUSSION THREAD

**Discussion by SasquatchSecurity** [Security-Onion-Solutions/securityonion#6610] [Unsupported Versions]
Using Winlogbeat to ship logs (4103 - 4106) does not ingest into Elastic if the ELK Powershell module is not specified as a processor in the winlogbeat.yml configuration. If the processor is added to the config, the fields it produces do not match with a number of the Elastalerts created via Playbook. 

Example field using ELK Module: powershell.file.script_block_text
Example field specified in play "Invoke-Obfuscation Obfuscated IEX Invocation": ScriptBlockText.keyword

It's clear that the ELK Module is renaming the field but due to events not being ingested without it, I'm not sure what to do moving forward. I do not see any events with ID 4104 or 4103 without the module enabled. With it enabled, 4103s still do not come through despite being in local Windows Event logs.

I believe the real "fix" is to let SO do the parsing of the Powershell logs, but I cannot seem to understand why the logs are not ingested without the module being enabled. I do not see a parser in /opt/so/conf/elasticsearch/ingest/ that would make sense for Powershell, so I assume that's why ELK doesn't know what to do with Powershell Operational logs. Looking for some assistance there. Other than changing the ELK module to map the field names correctly, I do not see a way around it.

Thanks!

From winlogbeat-powershell.js:

```javascript
    var addScriptBlockText = function (evt) {
        var text = evt.Get("winlog.event_data.ScriptBlockText");
        evt.Delete("winlog.event_data.ScriptBlockText");
        if (!text) {
            return;
        }

        evt.Put("powershell.file.script_block_text", text);
    };
```

**Comment by weslambert** [comment]
It's possible the mappings for Powershell script tblock logging might need to be updated from the Playbook side, but as far as ingestion, are you not seeing the events ingested into Elasticsearch at all? It might be worth it to tail the Logstash log on the search node(s) as you are testing to see if there is an issue with the events/records being indexed.
